lib/krb5/krbhst.c

```diff
@@ -795,7 +795,9 @@ kdc_get_next(krb5_context context,
 	return KRB5_KDC_UNREACH;
     }
 
-    if(context->srv_lookup) {
+    if (krb5_config_get_bool(context, NULL, "libdefaults", "block_dns",
+	    NULL) &&
+	context->srv_lookup) {
```

Comment:
Why are we running the following code only if `block_dns` is `true`? Correct me if I’m wrong, but shouldn’t these conditions be the other way round?

Reply:
I’ll submit a PR to fix this.